### PR TITLE
chain_node refactoring

### PR DESCRIPTION
**Pull request by rustaem**
Draft

### CODE REVIEW

File: apps/aecore/src/aec_consensus.erl [hunk 1]
```diff
@@ -158,17 +158,25 @@
...
 -callback state_pre_transform_key_node(#node{}, aec_trees:trees()) -> aec_trees:trees() | no_return().
 %% State pre transformations on every microblock
 -callback state_pre_transform_micro_node(#node{}, aec_trees:trees()) -> aec_trees:trees() | no_return().
+=======
+-callback state_pre_transform_key_node_consensus_switch(#chain_node{}, #chain_node{}, #chain_node{}, aec_trees:trees()) -> aec_trees:trees() | no_return().
+%% State pre transformations on every keyblock
+-callback state_pre_transform_key_node(#chain_node{}, #chain_node{}, #chain_node{}, aec_trees:trees()) -> aec_trees:trees() | no_return().
+%% State pre transformations on every microblock
+-callback state_pre_transform_micro_node(#chain_node{}, #chain_node{}, #chain_node{}, aec_trees:trees()) -> aec_trees:trees() | no_return().
+>>>>>>> 2f92aaea... Refactoring, dialyzer coverage
```
Comment (cytadela8) [inline apps/aecore/src/aec_consensus.erl:179]: Rebase issue!!!

**Review by cytadela8 — COMMENTED** [review]
@rustaem You have commited with merge conflicts :/ Remember to git add after resolving a conflict.

Also please make sure that the tests are passing.

File: apps/aecore/src/aec_consensus.erl [hunk 2]
```diff
@@ -182,6 +189,20 @@
 -callback genesis_raw_header() -> aec_headers:key_header().
 -callback genesis_difficulty() -> key_difficulty().
 
+%% -------------------------------------------------------------------
+%% Keyblock creation
+%% -------------------------------------------------------------------
+%% Creates a fake key node used for creating new keyblocks - can be blocking
```
Comment (cytadela8) [inline apps/aecore/src/aec_consensus.erl:195]: This is a change from hyperchains that should not have been copied.

File: apps/aecore/src/aec_consensus.erl [hunk 3]
```diff
@@ -348,3 +369,11 @@ get_genesis_hash() ->
         Consensus ->
             Consensus
     end.
+
+%% Brings down the entire node!
+%% Tries to inform the user what went wrong using any means
+-spec config_assertion_failed(string(), string(), list()) -> ok | no_return().
```
Comment (cytadela8) [inline apps/aecore/src/aec_consensus.erl:375]: This is a change from hyperchains that should not have been copied.

File: apps/aecore/src/aec_fork_block_settings.erl [hunk 4]
```diff
@@ -103,6 +103,10 @@ pre_iris_map_ordering() ->
             MapOrdering
     end.
 
+-spec hc_staking_contract_file() -> string().
+hc_staking_contract_file() ->
```
Comment (cytadela8) [inline apps/aecore/src/aec_fork_block_settings.erl:107]: This is a change from hyperchains that should not have been copied.

**Review by cytadela8 — CHANGES_REQUESTED** [review]

**Review by cytadela8 — CHANGES_REQUESTED** [review]
aec_chain_node module is missing. Please make the changes same as for hyperchains so we won't get significant conflicts.

**Comment by rustaem** [comment]
> aec_chain_node module is missing. Please make the changes same as for hyperchains so we won't get significant conflicts.

aec_chain_node was introduced in ra-3856-hc-and-core-refactoring (PR#3637). Spent some time, but cannot see a realistically simple way to backport those changes here. Want me to backport it hard way or should we postpone until hc merging into master?

**Pull request closed without merging**